### PR TITLE
types: note context type, note this type, and note return is a double curried function

### PR DESCRIPTION
**Pull request by ChristianMurphy**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -5,14 +5,15 @@
  *
  * @param {string} key
  * @param {boolean} state
- * @param {*} ctx
- * @returns {() => Function}
+ * @param {Record<string, any>} [ctx]
```
Comment (ChristianMurphy) [inline index.js:8]: library appears to have some expectation that keys are strings

File: index.js [hunk 2]
```diff
@@ -5,14 +5,15 @@
...
+ * @returns {() => () => void}
```
Comment (ChristianMurphy) [inline index.js:9]: main function returns enter function, which itself returns the exit function, which returns void.

File: index.js [hunk 3]
```diff
@@ -5,14 +5,15 @@
...
  */
 export function stateToggle(key, state, ctx) {
   return enter
 
   /**
    * Bound enter function to set `state` as `key` on `ctx`.
+   * @this {Record<string, any>}
```
Comment (ChristianMurphy) [inline index.js:16]: library appears to have some expectation that keys are strings

File: package.json [hunk 4]
```diff
@@ -27,6 +27,7 @@
     "index.js"
   ],
   "devDependencies": {
+    "@types/tape": "^4.0.0",
```
Comment (ChristianMurphy) [inline package.json:30]: missing tape types were hiding exception with return type

File: index.js (hunk 1 above)
Comment (ChristianMurphy) [reply index.js:8]: https://www.typescriptlang.org/docs/handbook/utility-types.html#recordkeystype

File: package.json (hunk 4 above)
Comment (wooorm) [reply package.json:30]: oh that’s interesting

**Pull request merged**